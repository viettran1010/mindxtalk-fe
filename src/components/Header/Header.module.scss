.header {
  background-color: #000000;
  .menu {
    background-color: #000000;
    .menu-item {
      color: #ffffff;
    }
  }
}
