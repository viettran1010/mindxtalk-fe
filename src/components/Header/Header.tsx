import { Layout, Menu } from "antd";
import styles from "./Header.module.scss";
import classnames from "classnames/bind";
import { browserHistory } from "../../App";
import { routeConstants } from "../../constants";

const cx = classnames.bind(styles);
const Header = () => {
  const { Header } = Layout;
  return (
    <Layout>
      <Header className={cx("header")}>
        <Menu mode="horizontal" className={cx("menu")}>
          <Menu.Item
            className={cx("menu-item")}
            onClick={() => browserHistory.push("/")}
          >
            HOME
          </Menu.Item>
          <Menu.Item
            className={cx("menu-item")}
            onClick={() => browserHistory.push(routeConstants.ROUTE_LOGIN)}
          >
            LOGIN
          </Menu.Item>
        </Menu>
      </Header>
    </Layout>
  );
};
export default Header;
