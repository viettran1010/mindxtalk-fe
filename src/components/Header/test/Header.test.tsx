import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../index';
afterEach(() => {
  jest.clearAllMocks();
  cleanup();
});

describe('<Header />', () => {
  it('should render correctly', async () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
