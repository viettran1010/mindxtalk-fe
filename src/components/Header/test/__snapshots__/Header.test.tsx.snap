// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<section
  class="ant-layout"
>
  <header
    class="ant-layout-header header"
  >
    <ul
      class="ant-menu ant-menu-light menu ant-menu-root ant-menu-horizontal"
      role="menu"
    >
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child menu-item"
        role="menuitem"
      >
        HOME
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child menu-item"
        role="menuitem"
      >
        LOGIN
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="visibility: hidden; position: absolute; display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
  </header>
</section>
`;
